scipy/stats/tests/test_stats.py

```diff
@@ -1983,7 +1983,9 @@ def test_scale(self):
                                     np.array([2, np.nan, 2]) / 1.3489795)
                 assert_equal(stats.iqr(x, axis=1, scale=2.0, nan_policy='propagate'),
                              [1, np.nan, 1])
-                _check_warnings(w, RuntimeWarning, 6)
+                # Since NumPy 1.17.0.dev, warnings are no longer emitted by
```

Comment:
I realize that this has already been merged, but I would prefer to keep the check, just wrapped in `if numpy_version < '1.16.0':`

Reply:
My reason is that IQR is extremely sensitive to the numpy version number, so I would like to retain all the version-specific checking in the tests. I think it might be even better to do something more explicit like `_check_warnings(w, RuntimeWarning, 0 if numpy_version >= '1.16.0a' else 6)`.

Reply:
That's how I originally had the check, at least as:
```
n_want = 0 if numpy_version >= '1.17.0.dev' else 6
```
assuming they are not backporting the `nan` changes to the `1.16` branch

I don't have a preference either way

Reply:
If it's not too much to reinstate that, I think it will be preferable in the long run. I think versions will give this function a lot more grief in the future, and it's important to retain traceability. The development process still gives me shivers.